### PR TITLE
Week 1

### PR DESCRIPTION
**Pull request by jerminatorhits**

### CODE REVIEW

File: .lein-repl-history [hunk 1]
```diff
@@ -0,0 +1,61 @@
+*ns*
```
Comment (portatlas) [inline .lein-repl-history:1]: you could put this file and the other target files in .gitignore

File: README.md [hunk 2]
```diff
@@ -1 +1,36 @@
-# clojure-tic-tac-toe
\ No newline at end of file
+# tic-tac-toe
+
+FIXME: description
```
Comment (portatlas) [inline README.md:3]: I would remove or upload the generic placeholder info

File: src/session.clj [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+      (empty? s))))
+
+(defn parse-int [s]
+   (Integer. (re-find  #"\d+" s )))
+
+(defn session-prompt
+  [msg]
+  (display-text msg)
+  (let [input (read-line)]
+  (if (and (numeric? input) (and (>= (parse-int input) 1) (<= (parse-int input) 9)))
```
Comment (portatlas) [inline src/session.clj:26]: Validating whether the input is within a specific range could be extracted into its own function.

File: src/session.clj [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    (- (parse-int input) 1)
+    (recur "Invalid Input. Please enter a valid move: "))))
+
+(defn session-run
+  []
+  (println (session-welcome))
```
Comment (portatlas) [inline src/session.clj:32]: you have a display-text function, why not use that?

File: src/session.clj [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+          s (drop-while #(Character/isDigit %) s)]
+      (empty? s))))
+
+(defn parse-int [s]
+   (Integer. (re-find  #"\d+" s )))
+
+(defn session-prompt
+  [msg]
+  (display-text msg)
+  (let [input (read-line)]
```
Comment (portatlas) [inline src/session.clj:25]: I would wrap the read-line into a function similar to how you wrapped println into display-text.

File: test/board_test.clj [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+            [board :refer :all]))
+
+(deftest board-test
+  (testing "creates a fresh board"
+    (is (= [1 2 3 4 5 6 7 8 9] (create-board))))
+
+  (testing "updates board when given a location and mark"
+    (is (= ["X" 2 3 4 5 6 7 8 9] (update-board [1 2 3 4 5 6 7 8 9] 0 "X")))))
+
+;;  (testing "formats board into presentable string when given a board"
```
Comment (portatlas) [inline test/board_test.clj:12]: Is there a reason why this test if commented out?

File: test/board_test.clj [hunk 7]
```diff
@@ -0,0 +1,13 @@
+(ns board-test
+  (:require [clojure.test :refer [deftest testing is]]
+            [board :refer :all]))
+
+(deftest board-test
+  (testing "creates a fresh board"
+    (is (= [1 2 3 4 5 6 7 8 9] (create-board))))
```
Comment (portatlas) [inline test/board_test.clj:7]: You are referencing an empty board a few times. Perhaps we can define it once and use it through out the test suite.

File: .lein-repl-history (hunk 1 above)
Comment (jerminatorhits) [reply .lein-repl-history:1]: Done. Target files are just the ones with ".nameOfFile" right? Not target folder?

File: README.md (hunk 2 above)
Comment (jerminatorhits) [reply README.md:3]: Done

File: src/session.clj (hunk 3 above)
Comment (jerminatorhits) [reply src/session.clj:26]: Done thanks

File: src/session.clj (hunk 4 above)
Comment (jerminatorhits) [reply src/session.clj:32]: Done

File: src/session.clj (hunk 5 above)
Comment (jerminatorhits) [reply src/session.clj:25]: Ok. I have changed it to this:   
  (let [input (String->Number (read-line))]
  (if (validate-entry input)
    (- input 1)
    (recur "Invalid Input. Please enter a valid move: "))))

Should I combine reading input + validating into one function?

File: test/board_test.clj (hunk 6 above)
Comment (jerminatorhits) [reply test/board_test.clj:12]: I was debating whether it is the board's job to format the board or if it is the display's job.

File: test/board_test.clj (hunk 7 above)
Comment (jerminatorhits) [reply test/board_test.clj:7]: (create-board) is a fresh board. This first test is simply used to see if (create-board) outputs an empty board correctly

File: src/session.clj (hunk 5 above)
Comment (portatlas) [reply src/session.clj:25]: No I wouldn't combine them. you should wrap read-line into a function like prompt-input similar to what you did with display-text. That way you are coupled to read-line and can change the implementation later if needed.

**Pull request merged**